lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2407,35 +2414,23 @@ class SqlParser(private val ion: IonSystem) : Parser {
             rem.head.err("Invalid format for time string. Expected format is \"TIME [(p)] [WITH TIME ZONE] HH:MM:SS[.ddddd...][+|-HH:MM]\"",
                 PARSE_INVALID_TIME_STRING)
         }
-        var newTimeString = timeString
-        when(isTimeZoneSpecified) {
-            false -> tryLocalTimeParsing(timeString)
-            true -> try {
-                OffsetTime.parse(timeString, DateTimeFormatter.ISO_TIME)
-            } catch (e: DateTimeParseException) {
-                // The exception thrown here is because of the invalid time or time zone offset specified in the timestring.
-                // The valid time zone offsets are in the range of [-18:00 - 18:00]
-                if (timeWithoutTimeZoneRegex.matches(timeString)) {
-                    // Fall back on parsing a string without a time zone offset only if the offset is not specified.
-                    // Add default timezone offset in that case.
-                    tryLocalTimeParsing(timeString)
-                    newTimeString = timeString + DEFAULT_TIMEZONE_OFFSET.getOffsetHHmm()
-                }
-                else {
-                    rem.head.err(e.localizedMessage, PARSE_INVALID_TIME_STRING)
-                }
-            }
-        }
+        // For "TIME WITH TIME ZONE", if the time zone is not explicitly specified, we still consider it as valid.
+        // We will add the default time zone to it later in the evaluation phase.
+        if (!withTimeZone || timeWithoutTimeZoneRegex.matches(timeString))
+            tryTimeParsing (timeString, DateTimeFormatter.ISO_TIME, LocalTime::parse)
```

Comment:
Remove additional space.
```suggestion
            tryTimeParsing(timeString, DateTimeFormatter.ISO_TIME, LocalTime::parse)
```